src/Student.cpp

```diff
@@ -0,0 +1,64 @@
+#include "Student.hpp"
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+Student::Student(){
+    name = "x";
+    nick = "y";
+    gitHub = "none";
+    firecode = "none";
+}
+
+Student::Student(std::string StudentNick, 
+                 std::string StudentName, 
+                 std::string gitHubNick, 
+                 std::string firecodeNick) {
+   // static unsigned int id = 0;
+    this -> nick = StudentNick;
+    this -> name = StudentName;
+    this -> gitHub = gitHubNick;
+    this -> firecode = firecodeNick;
+}
+
+Student::Student(const Student &obj){
```

Comment:
You can have the same behavior when you put `= default;` in hpp after this function declaration.